sdks/java/extensions/sql/build.gradle

```diff
@@ -24,6 +24,7 @@ plugins {
 }
 applyJavaNature(
   automaticModuleName: 'org.apache.beam.sdk.extensions.sql',
+  shadowClosure: {},
```

Comment:
@lukecwik I added this so I can build a single jar to pass to Java workers for the xlang transform (see the `--experiment='jar_packages=./sdks/java/extensions/sql/build/libs/beam-sdks-java-extensions-sql-2.18.0-SNAPSHOT.jar' \` line in my "how to run" in the description), but it looks like it's causing `:sdks:java:extensions:sql:validateShadedJarDoesntLeakNonProjectClasses` to fail spectacularly. Is there some other way I can do this?

Reply:
I don't think we want to do this since normal SQL users don't want to be using an artifact meant for the expansion service. It seems as though we should have the expansion service gradle project build one uber artifact containing all the xlang transforms we want to vend.

We could have multiple shadowJar tasks and produce multiple artifacts along this line but this seems overly complex initially.

@chamikaramj Is there an initial strategy we are following for how we want to build xlang expansion service jars?

Reply:
We need two jars.

(1) Jar for expansion service that contains the transform classes that need to be expanded (this can be multiple jars as well, IO transforms just have to be in the class path to be picked up by the AutoService).
(2) Jar to be passed to the Java worker. For me, building a shadow Jar of java-harness (:sdks:java:harness:shadowJar) worked.

It probably makes sens to move expansion service to it's own Gradle module and support building a shadow jar with all in-built cross-language transforms in Beam and release that.